basics/uattributes.adoc

```diff
@@ -59,29 +59,57 @@ The following table defines attributes that are used for all message types:
 |`id`
 |link:uuid.adoc[UUID]
 |yes
-|A unique message identifier. The timestamp contained in the UUID is used as the message's creation time.
+a|
+[.specitem,oft-sid="req~up-attributes-id~1",oft-needs="impl,utest"]
+--
+Every message *MUST* have a unique message identifier. The timestamp contained in the UUID is used as the message's creation time.
+--
 
 |`priority`
 |link:qos.adoc[UPriority]
 |no
-|The QoS level that this message should be processed/delivered with. If not specified explicitly, the link:qos.adoc#default-priority[default priority level] is assumed.
-
+a|
+The QoS level that this message should be processed/delivered with. If not specified explicitly, the link:qos.adoc#default-priority[default priority level] is assumed.
+[.specitem,oft-sid="req~up-attributes-priority~1",oft-needs="impl,utest"]
+--
+* *MUST* have a priority attribute.
```

Comment:
just two lines above, we say that it does not need to have one, aren't we? I would also suggest to require the attribute to have a value instead of the message having the attribute. FMPOV a message always has a _priority_ attribute, the attribute just doesn't always have (to have) a value set explicitly, and thus assume a default value.

I would suggest something like:
```
[.specitem,oft-sid="req~up-attributes-priority~1",oft-needs="impl,utest"]
--
A consumer *MUST* assume a message to have the link:qos.adoc#default-priority[default priority] if not set explicitly.
--

Reply:
changed